### PR TITLE
Update cs109b image

### PR DESCRIPTION
**Pull request by jaguillette**
This PR updates the base image for cs109b with the requested dependencies. I've included a build script to make it a bit easier to build the image for the right platform. The conda installs worked eventually, so I'm going to set Chris Gumb up for testing on the non-EKS version and see if everything is good. I'll un-draft this when I hear that everything is good.

### CODE REVIEW

**Comment by jaguillette** [comment]
Image has been approved as working by Chris Gumb, so this can be pulled in for use in the course.

File: apps/fas-jupyter-econ1040/Dockerfile [hunk 1]
```diff
@@ -14,5 +14,10 @@ FROM jupyter/minimal-notebook:notebook-6.5.1
 COPY --from=build /usr/local/bin/fa /usr/local/bin
 COPY --from=build /usr/local/bin/tournament /usr/local/bin
 
+USER $NB_UID
+WORKDIR $HOME
+COPY --chown=$NB_UID:$NB_GID src/automata/* .
+COPY --chown=$NB_UID:$NB_GID src/axelrod.prefs .
```
Comment (arthurian) [inline apps/fas-jupyter-econ1040/Dockerfile:20]: Did you intend to include `fas-jupyter-econ1040` changes in this PR?

File: apps/fas-jupyter-cs109b/Dockerfile [hunk 2]
```diff
@@ -1,52 +1,86 @@
...
+
+# Install packages that can only be installed via pip
+# COPY requirements.txt .
+# RUN pip install --upgrade pip && \
+#     pip install --no-cache-dir -r requirements.txt
+
+# RUN rm requirements.txt
+
+COPY cs109b.yml .
+RUN conda env update --name base --file cs109b.yml
```
Comment (arthurian) [inline apps/fas-jupyter-cs109b/Dockerfile:77]: Nice work getting this to install cleanly. Had such a tough time with this last year, it's a relief to see it simplified!

**Review by arthurian — APPROVED** [review]
LGTM! 👍 

Note that we will still likely need to create a separate image for GPU-enabled access since this image is missing the Nvidia CUDA libraries that are needed to interface with the Nvidia driver on the host. We may be able to take advantage of the [Jupyter Stacks: GPU Enabled Notebooks](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html#gpu-enabled-notebooks) work for that instead of having to do it ourselves, as in years past.

File: apps/fas-jupyter-cs109b/buildImage.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# pip-compile --resolver=backtracking --output-file requirements.txt requirements.in
+
+# build the image, tagging with parent folder name
+
+echo "Building docker image for ${imageName}"
+docker build --platform linux/amd64 -t $imageName . 
+
+# Export the installed packages to a reference file
+echo "Exporting requirements for ${imageName}"
+docker run --rm --platform linux/amd64 $imageName conda env export -n base > environment.yml
```
Comment (Tanvez) [inline apps/fas-jupyter-cs109b/buildImage.sh:18]: Cool script!

**Review by Tanvez — APPROVED** [review]
Looks good to me!

**Pull request merged**